demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -1042,20 +1073,34 @@ static int subscribePublishLoop( NetworkContext_t * pNetworkContext )
             LogInfo( ( "Sending Publish to the MQTT topic %.*s.",
                        MQTT_EXAMPLE_TOPIC_LENGTH,
                        MQTT_EXAMPLE_TOPIC ) );
-            returnStatus = publishToTopic( &mqttContext );
+            returnStatus = publishToTopic( pMqttContext );
 
             /* Calling MQTT_ProcessLoop to process incoming publish echo, since
              * application subscribed to the same topic the broker will send
              * publish message back to the application. This function also
              * sends ping request to broker if MQTT_KEEP_ALIVE_INTERVAL_SECONDS
              * has expired since the last MQTT packet sent and receive
              * ping responses. */
-            mqttStatus = MQTT_ProcessLoop( &mqttContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
+            mqttStatus = MQTT_ProcessLoop( pMqttContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
 
-            if( mqttStatus != MQTTSuccess )
+            /* No data available from the transport interface may be due to network
+             * delays or low timeout passed to #MQTT_ProcessLoop. */
+            if( mqttStatus == MQTTNoDataAvailable )
```

Comment:
`MQTT_ProcessLoop()` can't return `MQTTNoDataAvailable`, it resets the status internally to `MQTTSuccess` so that the loop will continue. If it didn't, and continued the loop in both cases, then the process loop would almost always return `MQTTNoDataAvailable`, as it's unlikely the whole timeout was needed for receiving data; so, it would have the same return code as only the last iteration of the loop.

Reply:
Yea. That is a good catch. As discussed offline, we will be able to loop until all the expected packets are received.